applications/asset_tracker_v2/src/cloud/cloud_codec/cloud_codec.h

```diff
@@ -287,9 +356,36 @@ int cloud_codec_encode_data(struct cloud_codec_data *output,
 			    struct cloud_data_accelerometer *accel_buf,
 			    struct cloud_data_battery *bat_buf);
 
+/**
+ * @brief Encode UI data.
+ *
+ * @param[out] output string buffer for encoding result
+ * @param[in] ui_buf UI data to encode
+ * @return 0 on success or error code according to errno.h.
+ */
 int cloud_codec_encode_ui_data(struct cloud_codec_data *output,
 			       struct cloud_data_ui *ui_buf);
 
+/**
+ * @brief Encode a batch of cloud buffer data.
+ *
+ * @param[out] output string buffer for encoding result
+ * @param[in] gnss_buf GNSS data buffer
+ * @param[in] sensor_buf Sensor data buffer
+ * @param[in] modem_stat_buf static modem data buffer
+ * @param[in] modem_dyn_buf dynamic modem data buffer
+ * @param[in] ui_buf button data buffer
+ * @param[in] accel_buf accelerometer data buffer
+ * @param[in] bat_buf battery data buffer
+ * @param[in] gnss_buf_count length of GNSS data buffer
+ * @param[in] sensor_buf_count length of Sensor data buffer
+ * @param[in] modem_stat_buf_count length of static modem data buffer
+ * @param[in] modem_dyn_buf_count length of dynamic modem data buffer
+ * @param[in] ui_buf_count length of button data buffer
+ * @param[in] accel_buf_count length of accelerometer data buffer
+ * @param[in] bat_buf_count length of battery data buffer
+ * @return 0 on success or error code according to errno.h.
```

Comment:
I would be nice if each individual return code could be documented, if possible with multiple backends. The return codes should be more or less aligned. That goes for the other APIs as well.